providers/common/include/internal/ciphers/ciphercommon.h

```diff
@@ -39,20 +39,20 @@ struct prov_cipher_ctx_st {
         ctr128_f ctr;
     } stream;
 
+    int enc;              /* Set to 1 for encrypt, or 0 otherwise */
+    size_t mode;
```

Comment:
Why?

An `unsigned char` would more than hold enough values for our lifetimes...

Reply:
reason explained above